symphony-bdk-template/symphony-bdk-template-freemarker/src/main/java/com/symphony/bdk/template/freemarker/FreeMarkerEngine.java

```diff
@@ -8,53 +8,31 @@
 import freemarker.template.TemplateExceptionHandler;
 import org.apache.commons.io.FilenameUtils;
 import org.apiguardian.api.API;
-import org.reflections.Reflections;
-import org.reflections.scanners.ResourcesScanner;
 
 import java.io.File;
 import java.io.IOException;
-import java.util.Map;
-import java.util.Set;
-import java.util.function.Function;
-import java.util.stream.Collectors;
 
 /**
  * FreeMarker specific implementation of {@link TemplateEngine}. Instantiates {@link FreeMarkerTemplate} objects.
+ *
+ * <p>
+ *   This class is thread-safe.
+ * </p>
  */
 @API(status = API.Status.INTERNAL)
 public class FreeMarkerEngine implements TemplateEngine {
 
-  public static final String FREEMARKER_EXTENSION = ".ftl";
-
-  private static Configuration configuration = createConfiguration();
-  private static Map<String, String> builtInTemplates = getBuiltInTemplateMap();
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public Set<String> getBuiltInTemplates() {
-    return builtInTemplates.keySet();
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public Template newBuiltInTemplate(String template) throws TemplateException {
-    if (!builtInTemplates.containsKey(template)) {
-      throw new TemplateException("Template " + template + " not found");
-    }
-    return newTemplateFromClasspath(builtInTemplates.get(template));
-  }
-
   /**
    * {@inheritDoc}
    */
   @Override
   public Template newTemplateFromFile(String templatePath) throws TemplateException {
     try {
-      return getTemplateFromFile(templatePath);
+      final String directory = FilenameUtils.getFullPathNoEndSeparator(templatePath);
+      final String file = FilenameUtils.getName(templatePath);
+      final Configuration configuration = createConfiguration(); // for thread-safety, we need to re-create configuration
```

Comment:
Thanks for putting this useful, because FreeMarker does not recommend to create a new configuration each time we load a template.

Reply:
The risk here was to update static configuration at method call level, it could be problematic in multi-threaded environment such as Spring Boot applications.